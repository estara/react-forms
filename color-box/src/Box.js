function Box(props) {
    const colors = {
      display: 'inline-block',
      width: `${props.width}em`,
      height: `${props.height}em`,
      backgroundColor: `${props.backgroundColor}`,
    };

    const remove = () => props.handleRemove(props.id);
  
    return (<div>
        <div style={colors}></div>
        <button onClick={remove}>X</button>
    </div>)
  }

  export default Box;