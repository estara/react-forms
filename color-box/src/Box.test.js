import { render, screen } from '@testing-library/react';
import Box from './Box';

test('renders without crashing', () => {
  render(<Box key='1' 
    id='1'
    handleRemove='removeBox' 
    width='5' 
    height='5'
    backgroundColor='green'/>);
});

it("matches snapshot", function() {
    const {asFragment} = render(<Box />);
    expect(asFragment()).toMatchSnapshot();
});