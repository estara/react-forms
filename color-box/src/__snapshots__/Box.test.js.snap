// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      style="display: inline-block;"
    />
    <button>
      X
    </button>
  </div>
</DocumentFragment>
`;
